packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +292,107 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed'] &&
+        @scenario_answers['foreman_proxy']['dns_provider'] == 'nsupdate'
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def answers_dns_values
+      new_vals = OpenStruct.new
+
+      new_vals.nameserver_ip = @scenario_answers['foreman_proxy']['dns_server']
+      new_vals.zone = @scenario_answers['foreman_proxy']['dns_zone']
+      new_vals.key_file = @scenario_answers['foreman_proxy']['keyfile']
+
+      new_vals.reverse_zones = [@scenario_answers['foreman_proxy']['dns_reverse']].flatten
+      run_cmd('rndc thaw')
```

Comment:
Thawing while determining answers feels like a weird side effect. I'd move this into `update_dns_records` before calling `nsupdate_command`.